### PR TITLE
Merge `Remapping` stencils and add 2D fields

### PR DESCRIPTION
**Pull request by eddie-c-davis**
This PR refactors several stencils to either merge them using horizontal regions or replace `numpy` code between stencils by applying lower dimensional fields. The specific changes include:

1. Add lower dimensional fields and typing information to `map_single` stencils.
2. Refactor tracer list and add typing to `MapN_Tracer2d`.
3. Change default version from `transliterated` to `stencil` in `mapn_tracer`.
4. Relax error constraints in `MapN_Tracer_2d` and `Remapping` accordingly
5. Add typing and some cleanup to `MoistCV`.
6. Changes to `remapping_part1`:
* Merge `moist_pt`, `delz_adjust`, and `pressure_updates` into `moist_cv_pt_pressure`.
* Merge `pn2_and_pk` and `copy_j_adjacent` into `pn2_and_pk`.
* Merge `undo_delz_adjust` and `peln` field copy into `undo_delz_adjust_and_copy_peln`.

### CODE REVIEW

File: fv3core/stencils/map_single.py [hunk 1]
```diff
@@ -297,114 +312,113 @@ def lagrangian_contributions_transliterated(
...
+                                    )
+                                )
+                                k0 = mm
+                                flag = 1
+                                elems[ii - i1, k2] = 0
+                                break
+                        # Add everything up and divide by the pressure difference
+                        q2[ii, j_2d, k2] = qsum / (pe2[ii, 0, k2 + 1] - pe2[ii, 0, k2])
+                        break
+    """
```
Comment (oelbert) [inline fv3core/stencils/map_single.py:424]: Is it worth it keeping this commented out code? I feel like it could be time to say goodbye and cut this out.

Comment (eddie-c-davis) [reply fv3core/stencils/map_single.py:424]: Could not agree more!

File: fv3core/stencils/remapping_part1.py [hunk 2]
```diff
@@ -23,21 +36,77 @@ def init_pe2(pe: sd, pe2: sd, ptop: float):
 
 
 @gtstencil()
-def delz_adjust(delp: sd, delz: sd):
+def undo_delz_adjust(delp: FloatField, delz: FloatField):
```
Comment (oelbert) [inline fv3core/stencils/remapping_part1.py:39]: Does this get used anywhere? Or just `undo_delz_adjust_and_copy_peln`?

**Review by oelbert — CHANGES_REQUESTED** [review]
Just the one question with undo_delz_adjust, but good stuff otherwise!

File: fv3core/stencils/remapping_part1.py (hunk 2 above)
Comment (eddie-c-davis) [reply fv3core/stencils/remapping_part1.py:39]: Nice catch!

**Comment by eddie-c-davis** [comment]
Good to go, thank you for the review @oelbert!

**Review by oelbert — APPROVED** [review]
Awesome. Excited to see the new timings

**Pull request merged**